libsolidity/formal/SymbolicVariables.cpp

```diff
@@ -171,27 +175,41 @@ SymbolicMappingVariable::SymbolicMappingVariable(
 	string _uniqueName,
 	EncodingContext& _context
 ):
-	SymbolicVariable(move(_type), move(_uniqueName), _context)
+	SymbolicVariable(_type, _type, move(_uniqueName), _context)
 {
 	solAssert(isMapping(m_type->category()), "");
 }
 
 SymbolicArrayVariable::SymbolicArrayVariable(
 	solidity::TypePointer _type,
+	solidity::TypePointer _originalType,
 	string _uniqueName,
 	EncodingContext& _context
 ):
-	SymbolicVariable(move(_type), move(_uniqueName), _context)
+	SymbolicVariable(_type, _originalType, move(_uniqueName), _context)
 {
 	solAssert(isArray(m_type->category()), "");
 }
 
+Expression SymbolicArrayVariable::currentValue(solidity::TypePointer const& _targetType) const
+{
+	if (_targetType)
+		// StringLiterals are encoded as SMT arrays in the generic case,
+		// but they can also be compared/assigned to fixed bytes, in which
+		// case they'd need to be encoded as numbers.
+		if (auto strType = dynamic_cast<StringLiteralType const*>(m_originalType))
+			if (_targetType->category() == solidity::Type::Category::FixedBytes)
+				return smt::Expression(u256(toHex(asBytes(strType->value()), HexPrefix::Add)));
```

Comment:
Any shortcut to this?

Reply:
Maybe ``import_bits`` works? https://www.boost.org/doc/libs/1_66_0/libs/multiprecision/doc/html/boost_multiprecision/tut/import_export.html

Reply:
hmm not sure, still need an `u256` in the end

Reply:
An ``u256`` is a boost multiprecision number - it should work as first argument to ``import_bits``...

Reply:
After talking to @ekpyron and @chriseth I'm convinced about the way it is at the moment. @ekpyron objections?

Reply:
Yeah - it feels a bit wasteful to first convert binary to hex only to have ``u256`` convert it back to binary again, but it's fine.